src/pages/RFOX/hooks/useCurrentEpochQuery.ts

```diff
@@ -0,0 +1,70 @@
+import { useQuery } from '@tanstack/react-query'
+import { RFOX_REWARD_RATE, RFOX_WAD } from 'contracts/constants'
+import dayjs from 'dayjs'
+import { queryClient } from 'context/QueryClientProvider/queryClient'
+
+import type { PartialEpoch } from '../types'
+import { getAffiliateRevenueQueryFn, getAffiliateRevenueQueryKey } from './useAffiliateRevenueQuery'
+import {
+  fetchCurrentEpochMetadata,
+  getCurrentEpochMetadataQueryKey,
+} from './useCurrentEpochMetadataQuery'
+import { fetchEpochHistory, getEpochHistoryQueryKey } from './useEpochHistoryQuery'
+
+type CurrentEpochMetadataQueryKey = ['currentEpoch']
+
+export const getCurrentEpochQueryKey = (): CurrentEpochMetadataQueryKey => ['currentEpoch']
+
+export const fetchCurrentEpoch = async (): Promise<PartialEpoch> => {
+  const currentEpochMetadata = await queryClient.fetchQuery({
+    queryKey: getCurrentEpochMetadataQueryKey(),
+    queryFn: fetchCurrentEpochMetadata,
+  })
+
+  const [historicalEpochs, affiliateRevenueRuneBaseUnit] = await Promise.all([
+    queryClient.fetchQuery({
+      queryKey: getEpochHistoryQueryKey(),
+      queryFn: fetchEpochHistory,
+    }),
+    queryClient.fetchQuery({
+      queryKey: getAffiliateRevenueQueryKey({
+        startTimestamp: BigInt(currentEpochMetadata.epochStartTimestamp),
+        endTimestamp: BigInt(currentEpochMetadata.epochEndTimestamp),
+      }),
+      queryFn: getAffiliateRevenueQueryFn({
+        startTimestamp: BigInt(currentEpochMetadata.epochStartTimestamp),
+        endTimestamp: BigInt(currentEpochMetadata.epochEndTimestamp),
+      }),
+    }),
+  ])
+
+  const lastHistoricalEpoch = historicalEpochs[historicalEpochs.length - 1]
+
+  // Calculate the total reward units for the current epoch so far. Note this is not the total
+  // reward units for the epoch because the epoch is still ongoing and the affiliate revenue is
+  // only complete up to the current block.
+  const secondsInCurrentEpoch = dayjs().unix() - currentEpochMetadata.epochStartTimestamp / 1000
+  const totalRewardUnits = (RFOX_REWARD_RATE / RFOX_WAD) * BigInt(secondsInCurrentEpoch)
```

Comment:
Comment above is key to this being correct.

See https://github.com/shapeshift/web/pull/7366/files#r1677338737